src/interfaces.ts

```diff
@@ -10,6 +10,8 @@ export interface BeamFocusableProps {
 export interface BeamButtonProps {
   /** Whether the interactive element is disabled. */
   disabled?: boolean;
+  /** If set, will show a tooltip about why the button is disabled. */
+  disabledReason?: string;
```

Comment:
FWIW: I had been musing with @KoltonG about whether it would be possible/good to have semi-direct integration with potential operations. Like:
```suggestion
  /** If a potential operation is provided, will show a tooltip about why the button is disabled when the operation is not allowed. */ 
  disabled?: boolean | PotentialOperation;
```

I think whether this is a good idea is questionable, since it would directly tie the tooltip content to the potential operation disabled reasons... when in most cases you'd probably want some additional processing of those reasons 🤔 

🤷

Reply:
Hrm... yeah, that would be pretty cute, would be a easy if the reasons were just strings:

```
type PotentialOperation2 {
  allowed: Boolean!
  disabledReasons: [Reason!]!
}

type Reason {
  message: String!
  entity: HasBlueprintUrl
}

interface HasBlueprintUrl {
  blueprintUrl: Url!
}
```

But since they're this complex `Reason` type, with pretty bespoke Blueprint-y things in it, yeah I'm not sure (as you already mentioned) we'd want to leak that directly into Beam...

Maybe if it was:

```
disabled: boolean | ReactNode
```

Where if `disabled` is every "not a boolean" / i.e. a `ReactNode` it knows that means "put the react node in the tooltip".

And then we had an adapter like:

```
<Button disabled=renderPotentialOperation(query.canDoFoo) />
```

Where the helper method would do app-specific rendering of the `PotentialOperation2` instead "just some react nodes to put into the tooltip"...

Reply:
@chr1sjf0x / @KoltonG pushed up another version that does:

```
disabled: boolean | ReactNode
```

Wdyt?

Reply:
Looks awesome!